client/src/partials/journal/modals/trialBalanceMain.body.js

```diff
@@ -213,40 +218,44 @@ function TrialBalanceMainBodyController(Session, trialBalanceService, Grouping,
     Notify.handleError(error);
   }
 
-  trialBalanceService.checkTransactions(vm.dataByTrans)
-    .then(function(error) {
-      var cssClass = null;
-      errorList = error.data;
-      vm.feedBack = trialBalanceService.getFeedBack(errorList); //getting a feedback object to customize the grid
-      vm.isInvalid = vm.feedBack.hasError || vm.feedBack.hasWarning;
-      cssClass = trialBalanceService.getCSSClass(vm.feedBack);
+  function startup() {
+    vm.loading = true;
 
-      $state.current.data.checkingData = {errors : errorList, feedBack : vm.feedBack, cssClass : cssClass};
-      $state.current.data.checked = true;
+    TrialBalance.checkTransactions(vm.dataByTrans)
+      .then(function (error) {
+        var cssClass = null;
 
-      columns.forEach(function (col) {
-        col.headerCellClass = cssClass;
-      });
+        errorList = error.data;
 
-      vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);
-    })
-    .catch(errorHandler)
-    .finally(function () {
-      vm.loading = false;
-      vm.showErrorButton = vm.isInvalid && !vm.loading;
-    });
+        vm.feedBack = TrialBalance.getFeedBack(errorList); // getting a feedback object to customize the grid
+        vm.isInvalid = vm.feedBack.hasError || vm.feedBack.hasWarning;
 
-  fetchDataByAccount()
-    .then(function (data) {
-      vm.gridOptions.data = data;
+        cssClass = TrialBalance.getCSSClass(vm.feedBack);
```

Comment:
Is there a reason these CSS classes can't be assigned in the template like in many other modules using `ng-class`? I haven't looked into the specifics but this logic is very specific and unknown to me. If it is required it would be a good thing to standardise for these cases.